llvm/include/llvm/IR/IntrinsicsAArch64.td

```diff
@@ -3808,50 +3808,50 @@ let TargetPrefix = "aarch64" in {
   def int_aarch64_sve_sel_x4  : SVE2_VG4_Sel_Intrinsic;
 
   class SME_LDR_STR_ZT_Intrinsic
-    : DefaultAttrsIntrinsic<[], [llvm_i32_ty, llvm_ptr_ty]>;
+    : DefaultAttrsIntrinsic<[], [llvm_i32_ty, llvm_ptr_ty], [IntrInaccessibleMemOrArgMemOnly]>;
```

Comment:
With the new modelling has the common base class lost its value? given we'll want to mark one as write only and the other as read only.

Reply:
Oh yes, of course.  I completely forgot about the actual load/store part of the intrinsic.